File: node_backend/middleware/request_logger.js
const REQUEST_LOG_PREFIX = '[node_backend]';

export function request_logger(request, response, next) {
  const start_time = Date.now();

  response.on('finish', () => {
    const duration_ms = Date.now() - start_time;

    console.info(`${REQUEST_LOG_PREFIX} ${request.method} ${request.originalUrl} ${response.statusCode} ${duration_ms}ms`);
  });

  next();
}

File: python_backend/middleware/requestLogger.py
from time import perf_counter

REQUEST_LOG_PREFIX = "[python_backend]"


def registerRequestLogger(app):
    @app.middleware("http")
    async def writeRequestLog(request, call_next):
        requestStartTime = perf_counter()
        response = await call_next(request)
        durationMs = (perf_counter() - requestStartTime) * 1000

        print(f"{REQUEST_LOG_PREFIX} {request.method} {request.url.path} {response.status_code} {durationMs:.2f}ms")
        return response
